agent/base

```diff
@@ -100,15 +100,30 @@ fi
 # don't use colons and dashes in the date suffix
 export date_suffix=$(date --date ${date} --utc +"%Y.%m.%dT%H.%M.%S")
 
+function validate_hostname {
+    validate-hostname "${1}"
+    if [[ ${?} -ne 0 ]]; then
```

Comment:
Not a suggestion (necessarily), just a question:  why use this idiom over the following?
```
    if ! validate-hostname "${1}"; then
```

Reply:
Creature of habit, my lack of deep understanding of `bash`, and I have always disliked executing commands in `if` statements.  That may be why I would like to convert all this to Python3 ... ;)

Reply:
I find that referencing `${?}` tends to be fragile (e.g., the temptation to insert debugging statements between the command which sets it and the code which tests it is overpowering at times, and, of course, doing so is catastrophic), while the idiom of immediately capturing the value in a local variable to prevent that problem tends to be "wordy" and cumbersome.  So, in many cases, invoking the command _in_ the `if` statement is just the right thing.

Don't think of it as "executing a command" -- think of it as making a subroutine call.  ;-)